zcutil/release-notes.py

```diff
@@ -0,0 +1,45 @@
+import re, sys, os
+import subprocess
+import argparse
+from itertools import islice
+
+# pulls list of authors from release notes
+def authors_in_release_notes():
+    with open(dir + filename, 'r') as f:
+        authors = []
+        first = f.readline().replace(':\n', '')
+        authors.append(first)
+        for line in f:
+            if line in ['\n', '\r\n']:
+                for a in islice(f, 1):
+                    a = a.replace(':\n', '')
+                    authors.append(a)
+        return authors
+
+## Documents release contributors in authors.md
+def document_authors():
+    with open('../doc/authors.md', 'a') as f:
+        f.write('\nContributors to {0}:'.format(version) + '\n')
+        print filename
+        for author in authors_in_release_notes():
+            print author
+            f.write(author + '\n')
+
+def main(version):
+    print version
+    p = subprocess.Popen(['git shortlog --no-merges v{0}..HEAD >../doc/release-notes/release-notes-{0}.md'.format(version)], shell=True, stdin=subprocess.PIPE, stdout=subprocess.PIPE)
+    stdin, stdout = p.communicate()
+    document_authors()
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser()
+    parser.add_argument('--version')
+    args = parser.parse_args()
+
+    dir = '../doc/release-notes/'
+    version = 'Please specify release version'
+    filename = None
```

Comment:
I'm not particularly fond of using a global for one of these and a method argument for the other. `filename` should be passed through.